controller/cache/cache.go

```diff
@@ -184,16 +184,20 @@ func (c *liveStateCache) loadCacheSettings() (*cacheSettings, error) {
 	return &cacheSettings{clusterSettings, appInstanceLabelKey}, nil
 }
 
-func asResourceNode(r *clustercache.Resource) appv1.ResourceNode {
+func asResourceNode(r *clustercache.Resource, c clustercache.ClusterCache) appv1.ResourceNode {
 	gv, err := schema.ParseGroupVersion(r.Ref.APIVersion)
 	if err != nil {
 		gv = schema.GroupVersion{}
 	}
 	parentRefs := make([]appv1.ResourceRef, len(r.OwnerRefs))
 	for _, ownerRef := range r.OwnerRefs {
 		ownerGvk := schema.FromAPIVersionAndKind(ownerRef.APIVersion, ownerRef.Kind)
-		ownerKey := kube.NewResourceKey(ownerGvk.Group, ownerRef.Kind, r.Ref.Namespace, ownerRef.Name)
-		parentRefs[0] = appv1.ResourceRef{Name: ownerRef.Name, Kind: ownerKey.Kind, Namespace: r.Ref.Namespace, Group: ownerKey.Group, UID: string(ownerRef.UID)}
+		var namespace string
+		if isNamespaced, _ := c.IsNamespaced(ownerGvk.GroupKind()); isNamespaced {
+			namespace = r.Ref.Namespace
+		}
```

Comment:
I was just wondering if we need to check the namespace of the resource being allowed by the AppProject of the owning application? While a namespace-scoped object only refers to an owner in the same namespace, this check was not required. But a cluster-scoped owner can be referred to from any namespace, and the application managing the owner might not have access to those namespaces. 

Would it be wise to display the dependents in non-allowed namespaces then?